tests/unit/backend/recipe/algorithm/test_LoadGroupingDefinition.py

```diff
@@ -249,25 +251,33 @@ def do_test_load_with_instrument_file(self, ext: str):
         loadingAlgo.setProperty("OutputWorkspace", outputWorkspace)
         assert loadingAlgo.execute()
         assert mtd.doesExist(outputWorkspace)
-        assert_wksp_almost_equal(outputWorkspace, self.localReferenceWorkspace)
+        assert workspacesEqual(outputWorkspace, self.localReferenceWorkspace)
         # check the function calls made
         calls = [call[0] for call in loadingAlgo.mantidSnapper._algorithmQueue]
         # check used correct cals
         for call in self.callsForExtension[ext]:
             assert call in calls
 
-    # NOTE LOADING FROM NAME IS VERY SLOW
     def do_test_load_with_instrument_name(self, ext: str):
+        # NOTE must get mantid to find the test instrument by its name
+        # documentation on mantid suggests this should be as easy as using the amend_config
+        # context manager to point into the directory with the test instrument, after adding
+        # a facilities.xml file.  For some reason, it just isn't working.
         outputWorkspace = "test_ext"
-        loadingAlgo = LoadingAlgo()
-        loadingAlgo.initialize()
-        loadingAlgo.mantidSnapper.cleanup = mock.Mock()
-        loadingAlgo.setProperty("GroupingFilename", self.localGroupingFile[ext])
-        loadingAlgo.setProperty("InstrumentName", "SNAP")
-        loadingAlgo.setProperty("OutputWorkspace", outputWorkspace)
-        assert loadingAlgo.execute()
+        with amend_config(**{"instrumentDefinition.directory": Resource.getPath("inputs/testInstrument/")}):
+            ConfigService.updateFacilities("inputs/testInstrument/Facilities.xml")
+            ConfigService.setFacility("TestSNAP")
+            loadingAlgo = LoadingAlgo()
+            loadingAlgo.initialize()
+            loadingAlgo.mantidSnapper.cleanup = mock.Mock()
+            loadingAlgo.setProperty("GroupingFilename", self.localGroupingFile[ext])
+            loadingAlgo.setProperty("InstrumentName", "fakeSNAP")
+            loadingAlgo.setProperty("OutputWorkspace", outputWorkspace)
+            assert loadingAlgo.execute()
+        ConfigService.updateFacilities("")
+        ConfigService.reset()
```

Comment:
This enables mantid to find the test instrument by name, then undoes all the changes so the rest of mantid works.